src/hardware/setup.rs

```diff
@@ -315,14 +316,14 @@ pub fn setup(
 
     // Load initial main-board settings from EEPROM
     let eeprom_settings = BoosterMainBoardData::load(&mut eeprom);
-    runtime_settings.fan_speed = eeprom_settings.fan_speed;
+    *runtime_settings.fan_speed = eeprom_settings.fan_speed;
 
     let mut settings = crate::settings::Settings {
         mac: eeprom_settings.mac,
-        ip: eeprom_settings.ip,
-        broker: eeprom_settings.broker,
-        gateway: eeprom_settings.gateway,
-        id: eeprom_settings.id,
+        ip: Leaf::from(eeprom_settings.ip),
+        broker: StrLeaf::from(eeprom_settings.broker),
+        gateway: Leaf::from(eeprom_settings.gateway),
+        id: StrLeaf::from(eeprom_settings.id),
```

Comment:
No `StrLeaf` here. Did you test that?

Reply:
It's a `StrLeaf<String<23>>`.

Reply:
But why? That's not the purpose of `StrLeaf` unless I'm missing something.

Reply:
Okay, I misunderstood the purpose then. I assumed it is a special case for types that implement `AsRef<str>`.